plugins/catalog-backend/src/ingestion/CatalogRules.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Config } from '@backstage/config';
+import { LocationSpec, Entity } from '@backstage/catalog-model';
+
+/**
+ * A structure for matching entities to a given rule.
+ */
+type EntityMatcher = {
+  kind: string;
+};
+
+/**
+ * A structure for matching locations to a given rule.
+ */
+type LocationMatcher = {
+  target?: string;
+  type: string;
+};
+
+/**
+ * Rules to apply to catalog entities
+ *
+ * An undefined list of matchers means match all, an empty list of matchers means match none
+ */
+type CatalogRule = {
```

Comment:
i was thinking more along the lines of 

```ts
type CatalogRule = {
  matcher: (entity: Entity) => boolean;
  action: 'allow' | 'reject';
};
```

since that's all that the rules engine needs to know. The constructor could take a `defaultAction: 'allow' | 'reject'` too I guess, but as mentioned elsewhere I would consider it an error if no matching rule was found (log harshly first time and reject, or even just explode?)

Reply:
Opted for not going with this to be able to provide more informative error messages. Either way it's an internal implementation and we can switch it out later without much trouble if we need to.